packages/interchain-token-service-api/src/primitives.rs

```diff
@@ -0,0 +1,87 @@
+use crate::error::Error;
+use cosmwasm_schema::cw_serde;
+use cosmwasm_std::{HexBinary, Uint256};
+use error_stack::Report;
+use router_api::Address;
+
+#[cw_serde]
+#[derive(Eq)]
+pub struct TokenId(
+    #[serde(with = "axelar_wasm_std::hex")]
+    #[schemars(with = "String")]
+    [u8; 32],
+);
+
+#[cw_serde]
+#[derive(Eq, Copy)]
+#[repr(u8)]
+pub enum TokenManagerType {
+    NativeInterchainToken = 0,
+    MintBurnFrom = 1,
+    LockUnlock = 2,
+    LockUnlockFee = 3,
+    MintBurn = 4,
+    Gateway = 5,
+}
+
+#[cw_serde]
+#[derive(Eq)]
+pub enum ITSMessage {
+    InterchainTransfer {
+        token_id: TokenId,
+        source_address: HexBinary,
+        destination_address: HexBinary,
+        amount: Uint256,
+        data: HexBinary,
+    },
+    DeployInterchainToken {
+        token_id: TokenId,
+        name: String,
+        symbol: String,
+        decimals: u8,
+        minter: HexBinary,
+    },
+    DeployTokenManager {
+        token_id: TokenId,
+        token_manager_type: TokenManagerType,
+        params: HexBinary,
+    },
+}
+
+#[cw_serde]
+#[derive(Eq)]
+pub struct ITSRoutedMessage {
+    /// Remote chain name.
+    /// ITS edge source contract -> ITS Hub GMP call: Set to the true destination chain name.
+    /// ITS Hub -> ITS edge destination contract: Set to the true source chain name.
+    pub remote_chain: Address,
```

Comment:
as CJ mentioned, this is really weird behaviour

Reply:
https://github.com/axelarnetwork/axelar-amplifier/pull/482#discussion_r1686782811